operations/compare-helm-with-jsonnet/components/config/kustomization.yaml

```diff
@@ -86,6 +86,14 @@ patches:
       - op: remove
         path: /config/frontend_worker
 
+  - target:
+      kind: MimirConfig
+      name: 'alertmanager|compactor|distributor|overrides-exporter|querier|query-frontend|query-scheduler|ruler|store-gateway'
+    patch: |-
+      # Jsonnet configures the ingester ring heartbeat period only on the ingester (other components don't need it).
```

Comment:
it's also set on the querier, ruler, and distributor

Reply:
which makes me wonder if we shouldn't be setting this on all components that use any ring (so set store-gateway ring config on queriers too). A quick and messy lookup makes me think that the timeout is enforced at the client (querier), not at the component partaking in the ring 


https://github.com/grafana/mimir/blob/bb8c6a45ce66d87ccb228de260ec3106dab18fdd/vendor/github.com/grafana/dskit/ring/model.go#L141

Reply:
I'm a bit lost.

The line you linked is `heartbeat_period` (how frequently the lifecycler should heartbeat the ring) and not the `heartbeat_timeout`. The `heartbeat_period` is only used in the lifecycler (so the component joining the ring), while `heartbeat_timeout` is used in the ring client to check which instances are healthy.

Reply:
i overlooked what was being ignored here. Sorry. You're right the timeout is being set in all components that use the ring in any way and the period only concerns the component which runs the lifecycler.